public/app/features/alerting/unified/components/rule-viewer/tabs/components/VersionHistoryTable.tsx

```diff
@@ -0,0 +1,190 @@
+import { useState } from 'react';
+
+import { dateTimeFormat, dateTimeFormatTimeAgo } from '@grafana/data';
+import { config } from '@grafana/runtime';
+import { Badge, Button, Checkbox, Column, ConfirmModal, InteractiveTable, Stack, Text } from '@grafana/ui';
+import { Trans, t } from 'app/core/internationalization';
+import { computeVersionDiff } from 'app/features/alerting/unified/utils/diff';
+import { DiffGroup } from 'app/features/dashboard-scene/settings/version-history/DiffGroup';
+import { Diffs, jsonDiff } from 'app/features/dashboard-scene/settings/version-history/utils';
+import { GrafanaRuleDefinition, RulerGrafanaRuleDTO } from 'app/types/unified-alerting-dto';
+
+import { UpdatedByUser } from './UpdatedBy';
+
+const VERSIONS_PAGE_SIZE = 20;
+
+export function VersionHistoryTable({
+  onVersionsChecked,
+  ruleVersions,
+  disableSelection,
+}: {
+  onVersionsChecked: (value: Map<string, boolean>) => void;
+  ruleVersions: Array<RulerGrafanaRuleDTO<GrafanaRuleDefinition>>;
+  disableSelection: boolean;
+}) {
+  const [checkedVersions, setCheckedVersions] = useState<Map<string, boolean>>(new Map());
```

Comment:
`AlertVersionHistory` also keeps `checkedVersions` state. Should we try to pass `checkedVersions` as a prop instead? Currently, we need to maintain the same state in two different places

Also, should we consider using a tuple for storing the checked versions? The user cannot select more than two versions. Alternatively, maybe we should invoke the `onVersionsChecked` callback only, if the user selected two versions? (Then `checkedVersions` state management should not be necessary in the `AlertVersionHistory` component

Reply:
done!